tests/unit/test_client.py

```diff
@@ -68,7 +68,7 @@ def test_request_headers(mock_get_and_post):
     client_info_header = constants.HEADER_CLIENT_INFO
     client_info_value = "some_client_info"
 
-    req = TrinoRequest(
+    req_without_role = TrinoRequest(
```

Comment:
Please split the tests per scenario. That makes it so much easier to troubleshoot if problems arise. Avoid numbering and try to make the name of the scenario as explicit as possible.